### PR TITLE
SpriteRenderer.destroy fix

### PR DESCRIPTION
**Pull request by staff0rd**
SpriteRenderer.destroy() was failing - looks related to #3201

### CODE REVIEW

**Comment by staff0rd** [comment]
Hmm, no webgl on travis CI.  Thoughts?

**Comment by englercj** [comment]
You could try just passing in a canvas renderer? Since you're not actually doing anything with it but destroying it.

**Comment by staff0rd** [comment]
mocked

File: src/core/sprites/webgl/SpriteRenderer.js [hunk 1]
```diff
@@ -452,13 +452,7 @@ export default class SpriteRenderer extends ObjectRenderer
...
         super.destroy();
 
-        for (let i = 0; i < this.shaders.length; i++)
-        {
-            if (this.shaders[i])
-            {
-                this.shaders[i].destroy();
-            }
-        }
+        this.shader.destroy();
```
Comment (englercj) [inline src/core/sprites/webgl/SpriteRenderer.js]: check for existence, construction and immediate destruction without context creation would make this error out.

Comment (bigtimebuddy) [inline src/core/sprites/webgl/SpriteRenderer.js]: Also, make sure to nullify. `this.shader = null`

**Comment by staff0rd** [comment]
I don't know how to get rid of the merge at the end - does it matter?

**Comment by englercj** [comment]
Nah, I'll squash it out when I merge.

File: src/core/sprites/webgl/SpriteRenderer.js (hunk 1 above)
Comment (staff0rd) [inline src/core/sprites/webgl/SpriteRenderer.js]: @englercj construction and immediate destruction without context creation was broken anyway - fixed with cc58b0e

**Pull request merged**